rust/agama-lib/src/storage/client/dasd.rs

```diff
@@ -21,39 +21,47 @@
 //! Implements a client to access Agama's D-Bus API related to DASD management.
 
 use zbus::{
-    fdo::ObjectManagerProxy,
+    fdo::{IntrospectableProxy, ObjectManagerProxy},
     zvariant::{ObjectPath, OwnedObjectPath},
     Connection,
 };
 
 use crate::{
     error::ServiceError,
-    storage::{model::dasd::DASDDevice, proxies::DASDManagerProxy},
+    storage::{model::dasd::DASDDevice, proxies::dasd::ManagerProxy},
 };
 
 /// Client to connect to Agama's D-Bus API for DASD management.
 #[derive(Clone)]
 pub struct DASDClient<'a> {
-    manager_proxy: DASDManagerProxy<'a>,
+    manager_proxy: ManagerProxy<'a>,
     object_manager_proxy: ObjectManagerProxy<'a>,
+    introspectable_proxy: IntrospectableProxy<'a>,
 }
 
 impl<'a> DASDClient<'a> {
     pub async fn new(connection: Connection) -> Result<DASDClient<'a>, ServiceError> {
-        let manager_proxy = DASDManagerProxy::new(&connection).await?;
+        let manager_proxy = ManagerProxy::new(&connection).await?;
         let object_manager_proxy = ObjectManagerProxy::builder(&connection)
             .destination("org.opensuse.Agama.Storage1")?
             .path("/org/opensuse/Agama/Storage1")?
             .build()
             .await?;
+        let introspectable_proxy = IntrospectableProxy::builder(&connection)
+            .destination("org.opensuse.Agama.Storage1")?
+            .path("/org/opensuse/Agama/Storage1")?
+            .build()
+            .await?;
+
         Ok(Self {
             manager_proxy,
             object_manager_proxy,
+            introspectable_proxy,
         })
     }
 
     pub async fn supported(&self) -> Result<bool, ServiceError> {
-        let introspect = self.manager_proxy.introspect().await?;
+        let introspect = self.introspectable_proxy.introspect().await?;
         // simply check if introspection contain given interface
         Ok(introspect.contains("org.opensuse.Agama.Storage1.DASD.Manager"))
```

Comment:
finally they split it to own proxy. Maybe code can be improved even check...but looks like they still have xml parsing separately. So no easy way to improve it.